libauk/src/main/java/com/bitmark/libauk/storage/SecureFileStorage.kt

```diff
@@ -63,8 +70,56 @@ internal class SecureFileStorageImpl(
         return os.toByteArray()
     }
 
-    override fun readOnFilesDir(name: String): ByteArray =
-        read(File(context.filesDir, "$alias-$name").absolutePath)
+    override fun readOnFilesDir(name: String): Single<ByteArray> {
+        var byteArray = byteArrayOf()
+        val isAuthenRequired = BiometricUtil.isAuthenReuired(listOf(name), context)
+        if (isAuthenRequired) {
+            if (context is FragmentActivity) {
+                return BiometricUtil.withAuthenticate<ByteArray>(activity = context,
+                    onAuthenticationSucceeded = { result ->
+                        read(
+                            File(context.filesDir, "$alias-$name").absolutePath,
+                            isAuthenRequired
+                        ).also { byteArray = it }
+                    },
+                    onAuthenticationError = { _, _ -> byteArrayOf() },
+                    onAuthenticationFailed = { byteArrayOf() }
+                )
+            }
+        }
+        else {
+            return Single.fromCallable { read(File(context.filesDir, "$alias-$name").absolutePath, isAuthenRequired) }
+        }
+        return Single.fromCallable { byteArray }
+    }
+
+    override fun readOnFilesDirWithoutAuthentication(name: String): ByteArray {
+        return read(File(context.filesDir, "$alias-$name").absolutePath, false)
+    }
+
+    override fun readFiles(names: List<String>): Single<Map<String, ByteArray>> {
+        val map = mutableMapOf<String, ByteArray>()
+        val isAuthenRequired = BiometricUtil.isAuthenReuired(names, context)
+        if (isAuthenRequired) {
+            BiometricUtil.withAuthenticate(
+                activity = context as FragmentActivity,
+                onAuthenticationSucceeded = { result ->
+                    names.forEach { name ->
+                        read(File(context.filesDir, "$alias-$name").absolutePath, isAuthenRequired).also { map[name] = it }
+                    }
+                    map
+                },
+                onAuthenticationError = { _, _ -> map },
+                onAuthenticationFailed = { map }
+            )
+        }
+        return Single.fromCallable {
+            names.forEach { name ->
+                read(File(context.filesDir, "$alias-$name").absolutePath, isAuthenRequired).also { map[name] = it }
+            }
+            map
+        }
+    }
```

Comment:
Caching on a map may not work because it will return before onAuthenticatedSuccess.

here is a suggested function:

```
override fun readFiles(names: List<String>): Single<Map<String, ByteArray>> {
    val isAuthenRequired = BiometricUtil.isAuthenReuired(names, context)

    fun readFileContents(): Map<String, ByteArray> = names.associateWith { name ->
        read(File(context.filesDir, "$alias-$name").absolutePath, isAuthenRequired)
    }

    return if (isAuthenRequired) {
        if (context is FragmentActivity) {
            BiometricUtil.withAuthenticate<Map<String, ByteArray>>(
                activity = context,
                onAuthenticationSucceeded = { readFileContents() },
                onAuthenticationError = { _, _ -> emptyMap() },
                onAuthenticationFailed = { emptyMap() } 
            )
        } else {
            Single.error(IllegalStateException("Context is not an instance of FragmentActivity"))
        }
    } else {
        Single.fromCallable { readFileContents() }
    }
}
```